### DISCUSSION TITLE
ModuleNotFoundError: No module named 'cv2'

### DISCUSSION THREAD

**Discussion by Tranthanhbao198** [NVlabs/curobo#312] [Software Q&A]
If it’s not a bug, please use discussions: https://github.com/NVlabs/curobo/discussions

Please provide the below information in addition to your issue:

1. cuRobo installation mode (choose from [python, isaac sim, docker python, docker isaac sim]): IsaacSim
2. python version: in conda env python 3.9
3. Isaac Sim version (if using): 2022.2.1
4.  2 GPUs Nvidia gerforce rtx 4090

------
hello, i got this error. I installed opencv-python both in conda env (PnP) and base, but i always got error with cv2 when i ran omni_python realsense_viewer.py. Please help me to fix it. thank you so much

**Comment by Tranthanhbao198** [comment]
<img width="1440" alt="Ảnh màn hình 2024-05-31 lúc 11 48 47" src="https://github.com/NVlabs/curobo/assets/75514870/180e2a45-94cb-4332-bd2d-417dd2dac0c3">

**Comment by balakumar-s** [comment]
Can you try: `omni_python -m pip install opencv-python`?

**Comment by Tranthanhbao198** [comment]
> Can you try: `omni_python -m pip install opencv-python`?
CMake Error at CMakeLists.txt:32 (find_package):
  By not providing "FindTorch.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Torch", but
  CMake did not find one.

  Could not find a package configuration file provided by "Torch" with any of
  the following names:

    TorchConfig.cmake
    torch-config.cmake

  Add the installation prefix of "Torch" to CMAKE_PREFIX_PATH or set
  "Torch_DIR" to a directory containing one of the above files.  If "Torch"
  provides a separate development package or SDK, be sure it has been
  installed.


-- Configuring incomplete, errors occurred!
hello, i am reinstalling, but at step8, i got this error, can you help me?
<img width="1440" alt="Ảnh màn hình 2024-05-31 lúc 14 20 50" src="https://github.com/NVlabs/curobo/assets/75514870/0a3fd871-7c3f-4272-bc8c-ca01d63a799e">

**Comment by Tranthanhbao198** [comment]
and this error

> Can you try: `omni_python -m pip install opencv-python`?

yes thank you it worked

**Comment by Tranthanhbao198** [comment]
I still got some error like that:
1. OSError: /home/apicoo/pkgs/nvblox_torch/src/nvblox_torch/bin/libpy_nvblox.so: cannot open shared object file: No such file or directory
2024-05-31 09:47:03 [23,610ms] [Warning] [carb.audio.context] 1 contexts were leaked
/home/apicoo/.local/share/ov/pkg/isaac_sim-2022.2.1/python.sh: line 41: 125594 Segmentation fault      (core dumped) $python_exe "$@" $args
There was an error running python
**_**-> this error is appeared when i run omni_python realsense_collision.py**_**
<img width="1440" alt="Ảnh màn hình 2024-05-31 lúc 16 54 17" src="https://github.com/NVlabs/curobo/assets/75514870/01276741-1a22-4e24-862f-31bc4817b375">


2. (PnP) apicoo@apicoo-pc:~/curobo/examples/isaac_sim$ omni_python realsense_viewer.py
Warning: running in conda env, please deactivate before executing this script
If conda is desired please source setup_conda_env.sh in your python 3.7 conda env and run python normally
Depth Scale is:  0.0010000000474974513
Traceback (most recent call last):
  File "realsense_viewer.py", line 34, in <module>
    view_realsense()
  File "realsense_viewer.py", line 11, in view_realsense
    data = realsense_data.get_raw_data()
  File "/home/apicoo/pkgs/nvblox_torch/src/nvblox_torch/datasets/realsense_dataset.py", line 193, in get_raw_data
    depth_np, rgb_np = get_next_depth_and_color(self.realsense, filter_color=False)
  File "/home/apicoo/pkgs/nvblox_torch/src/nvblox_torch/datasets/realsense_dataset.py", line 80, in get_next_depth_and_color
    frames = realsense["pipeline"].wait_for_frames()
RuntimeError: Frame didn't arrive within 5000
There was an error running python
**--> this code when i run omni_python realsense_viewer.py**
<img width="1440" alt="Ảnh màn hình 2024-05-31 lúc 16 54 17" src="https://github.com/NVlabs/curobo/assets/75514870/6effdf2d-7ee6-48e1-986f-60bdbe98a6e3">

**Comment by Tranthanhbao198** [comment]
3.  [14%] Built target ext_eigen
[ 15%] Building CUDA object CMakeFiles/nvblox_gpu_hash.dir/src/core/error_check.cu.o
[ 16%] Building CXX object CMakeFiles/nvblox_gpu_hash.dir/src/utils/nvtx_ranges.cpp.o
[ 17%] Building CXX object CMakeFiles/nvblox_gpu_hash.dir/src/utils/timing.cpp.o
[ 20%] Building CUDA object CMakeFiles/nvblox_gpu_hash.dir/src/gpu_hash/gpu_layer_view.cu.o
[ 20%] Building CUDA object CMakeFiles/nvblox_gpu_hash.dir/src/gpu_hash/gpu_set.cu.o
nvcc fatal   : Unsupported gpu architecture 'compute_89'
make[2]: *** [CMakeFiles/nvblox_gpu_hash.dir/build.make:135: CMakeFiles/nvblox_gpu_hash.dir/src/gpu_hash/gpu_set.cu.o] Error 1
make[2]: *** Waiting for unfinished jobs....
nvcc fatal   : Unsupported gpu architecture 'compute_89'
make[2]: *** [CMakeFiles/nvblox_gpu_hash.dir/build.make:77: CMakeFiles/nvblox_gpu_hash.dir/src/core/error_check.cu.o] Error 1
nvcc fatal   : Unsupported gpu architecture 'compute_89'
make[2]: *** [CMakeFiles/nvblox_gpu_hash.dir/build.make:120: CMakeFiles/nvblox_gpu_hash.dir/src/gpu_hash/gpu_layer_view.cu.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:261: CMakeFiles/nvblox_gpu_hash.dir/all] Error 2
make: *** [Makefile:136: all] Error 2

**--> this error when i run step 7 in Installing nvblox for PRECXX11_ABI and Isaac Sim**
<img width="1440" alt="Ảnh màn hình 2024-05-31 lúc 16 57 14" src="https://github.com/NVlabs/curobo/assets/75514870/cd40254b-1aad-4dd2-83ce-a6a1cd01f557">

**Comment by Tranthanhbao198** [comment]
I use 2 GPUs Nvidia gerforce rtx 4090

**Comment by Tranthanhbao198** [comment]
@balakumar-s bala
 please help me to fix them. thank you so much

**Comment by balakumar-s** [comment]
Is it possible for you to use the provided docker?

**Comment by balakumar-s** [comment]
In addition, python version in conda should match isaac sim python version of 3.8.

Also, check your cuda version. It should be 11.8.

**Comment by Tranthanhbao198** [comment]
> Is it possible for you to use the provided docker?

Hello, i dont have many experiment with docker@.@

**Comment by Tranthanhbao198** [comment]
> In addition, python version in conda should match isaac sim python version of 3.8.
> 
> Also, check your cuda version. It should be 11.8.
How do I know version of python isaacsim? Because I don't see any information about it before? Please let me know.
And, I read docs and it said that I should install cuda11.7, so I should need to reinstall it? 
Thank you so much.
